lib/driver.js

```diff
@@ -866,6 +886,42 @@ class XCUITestDriver extends BaseDriver {
     });
   }
 
+  /**
+   * Start preinstalled WDA process via XCTest.
+   *
+   * @param {string} sessionId A session id to proxy requests to.
+   * @return {Promise<object>} State Object
+   */
+  async startXctestWda(sessionId) {
+    const launchWDABundleID = this.opts.updatedWDABundleId || defaultWDARunnerBundleID;
+    this.mobileKillApp(launchWDABundleID);
+
+    if (!this.xctestWDA) {
+      const xctestEnv = {};
+      if (this.opts.wdaLocalPort) {
+        xctestEnv.USE_PORT = this.opts.wdaLocalPort;
+      }
+      this.xctestWDA = new Xctest(this.opts.udid, launchWDABundleID, null, {env: xctestEnv});
+    }
+    await this.xctestWDA.start();
+
+    this.wda.setupProxies(sessionId);
+    this.wda.fullyStarted = true;
+    return await this.wda.getStatus();
```

Comment:
I'll try to move this part in https://github.com/appium/WebDriverAgent (need to add appium-ios-device dependency tho) to wrap around this in `this.wda`